### ISSUE TITLE
sbt not sending authorization headers when bootstrapping

### ISSUE THREAD

**Issue by devshorts** [sbt/sbt#3799]
sbt version: 0.13.16
sbt-launcher version: 0.13.16/1.04

My company has an internal nexus installation that we use as a proxy. We have credentials configured in a few places. The plaintext creds are in `~/sbt/credentials`. We have in `/usr/local/etc/sbtopts` the following:

```
-Dscala.color=true -Dsbt.override.build.repos=true -Dsbt.boot.credentials=/Users/antonkropp/.sbt/credentials
```

Our repositories file `~/.sbt/repositories` looks like:

```
[repositories]
  local
  curalate-ivy-proxy-releases: https://maven.curalate.com/content/groups/ivy, [organization]/[module]/(scala_[scalaVersion]/)(sbt_[sbtVersion]/)[revision]/[type]s/[artifact](-[classifier]).[ext]
  curalate-maven-proxy-releases: https://maven.curalate.com/content/groups/omnibus
```

Resolving internal artifacts via our proxy during normal sbt builds works fine. The issue arises when we want to _boot_ to a new sbt version.  For the life of me I cannot get the sbt launcher to send credentials over the wire:

As an example, the http request that the sbt launcher (0.13.16) makes is:

```
HEAD /content/groups/ivy/com.curalate/curalate-macros_2.12/1.0-2-bde3695/ivys/ivy.xml HTTP/1.1
User-Agent: Apache Ivy/2.3.0-sbt-48dd0744422128446aee9ac31aa356ee203cc9f4
Host: maven.curalate.com
Accept: text/html, image/gif, image/jpeg, *; q=.2, */*; q=.2
Connection: keep-alive
```

With the proper 401 response of

```
HTTP/1.1 401 Unauthorized
Content-Length: 0
Date: Wed, 06 Dec 2017 21:15:06 GMT
Server: Nexus/3.6.0-02 (OSS)
WWW-Authenticate: BASIC realm="Sonatype Nexus Repository Manager"
X-Content-Type-Options: nosniff
X-Frame-Options: SAMEORIGIN
Connection: Keep-alive
```

Even though the realm we have set up in the credentials file is:

```
$ cat ~/.sbt/credentials
realm=Sonatype Nexus Repository Manager
host=maven.curalate.com
user=username
password=xXXXxxXX
```

I cloned the 0.13 branch and ran a debugger against it. 

```
java -Xrunjdwp:transport=dt_socket,server=y,suspend=y,address=5005 -jar /usr/local/Cellar/sbt@0.13/0.13.16/libexec/bin/sbt-launch.jar -Dsbt.boot.credentials=/Users/antonkropp/.sbt/credentials -Dsbt.override.build.repos=true -Xdebug
```

I can see that on boot

![image](https://user-images.githubusercontent.com/1799346/33687029-99bd7160-da8b-11e7-9dd3-064087e268f9.png)

If I evaluate `add` with the credentials file information I am able to resolve and download from my local nexus so the ivy proxy configuration of our local nexus is correct.

From just looking at the code though, I don't see where credentials gets set on boot:

![image](https://user-images.githubusercontent.com/1799346/33687063-ba9e0836-da8b-11e7-972a-fb92cc5db0a1.png)

Only _after_ bootstrapping has occurred do I see the credentials file being added:

![image](https://user-images.githubusercontent.com/1799346/33687132-fe6cd272-da8b-11e7-8450-c9205a09cb26.png)

Are there any suggestions for getting sbt to bootstrap itself? I've tried launchers of both sbt 0.13.16 and 1.0.4 and no credentials ever get picked up.

**Comment by yelianevich** [comment]
I haven't figured out how to send credentials during SBT update, that's something that should be addressed by sbt itself in my opinion.

Nevertheless I found a decent workaround for that. I specified in `~/.sbt/repositories` file 2 additional repositories in the end, that only get hit when internal nexus cannot resolve dependency. Your `~/.sbt/repositories` file will look as follows:
```
[repositories]
  local
  curalate-ivy-proxy-releases: https://maven.curalate.com/content/groups/ivy, [organization]/[module]/(scala_[scalaVersion]/)(sbt_[sbtVersion]/)[revision]/[type]s/[artifact](-[classifier]).[ext]
  curalate-maven-proxy-releases: https://maven.curalate.com/content/groups/omnibus
  public-ivy-releases: https://repo.typesafe.com/typesafe/ivy-releases/, [organization]/[module]/(scala_[scalaVersion]/)(sbt_[sbtVersion]/)[revision]/[type]s/[artifact](-[classifier]).[ext]
  maven-central
```
`public-ivy-releases` was added in order to resolve sbt 0.13.x and `maven-central` to update to sbt `1.x`

**Comment by eed3si9n** [comment]
Could you define:

```
export SBT_CREDENTIALS="$HOME/.sbt/credentials"
```

and see if that solve it plz?

**Comment by devshorts** [comment]
@eed3si9n wow, that worked. Thank you!!

**Comment by eed3si9n** [comment]
I'll close this as resolved since there's #3844 already to track for documentation.
